backend/typescript/services/implementations/camperService.ts

```diff
@@ -1228,6 +1265,85 @@ class CamperService implements ICamperService {
       throw error;
     }
   }
+
+  /* eslint-disable class-methods-use-this */
+  async getRefundInfo(refundCode: string): Promise<RefundDTO> {
+    let refundDTO: RefundDTO = [];
+    let refundCamper: RefundCamperDTO;
+    let campSession: CampSession | null = null;
+    let key: string;
+    let oldRefundInstance: RefundCamperGroupDTO | undefined;
+    let newRefundInstance: RefundCamperGroupDTO;
+    const refundDTOMap = new Map<string, RefundCamperGroupDTO>();
+    try {
+      const campers: Camper[] = await MgCamper.find({ refundCode });
+      if (!campers || campers.length === 0) {
+        throw new Error(`Campers with Refund Code ${refundCode} not found.`);
+      }
+
+      // need a camp session to get camp id to get start and end times from camp
+      campSession = await MgCampSession.findById(campers[0].campSession);
+      if (!campSession) {
+        throw new Error(
+          `Camp session with ID ${campers[0].campSession} not found.`,
+        );
+      }
+
+      const camp = await MgCamp.findById(campSession.camp);
+      if (!camp) {
+        throw new Error(`Camp with ID ${campSession.camp} not found.`);
+      }
+
+      // eslint-disable-next-line no-restricted-syntax
+      for (const camper of campers) {
+        // eslint-disable-next-line no-await-in-loop
+        campSession = await MgCampSession.findById(camper.campSession);
+        key = camper.firstName + camper.lastName;
+        refundCamper = {
```

Comment:
This is a lot of lines to copy data over. Maybe we should just keep all the fields of `Camper` (and change the DTO to reflect that)

Reply:
Agree, that's a great point and idea. I tried this but keeping all the fields of `camper` results in strange formatting of the data. Would propose to keep the copying for easy data handling on the frontend.